### ISSUE TITLE
panic on 0.5.3 when DNS lookup fails

### ISSUE THREAD

**Issue by boyvinall** [hashicorp/terraform#2403]
I know this is a corner-case, but someone else may run into this, and it'd be nice to get a cleaner error message. The virtualbox VM from which I was running terraform managed to lose it's ability to resolve DNS. (I could still ping internet hosts, just DNS had gone bad.) I was able to run "terraform plan" fine, but "terraform apply" gave me:
```
panic: runtime error: invalid memory address or nil pointer dereference
```
along with a big stack dump of each goroutine etc, and crash dump as per https://gist.github.com/boyvinall/fd65150b67862af13351

This was with the AWS provider. I'd reduced the config to just the following while trying to isolate the issue:
```
provider "aws" {
    region = "${var.aws_region}"
    access_key = "${var.aws_access_key}"
    secret_key = "${var.aws_secret_key}"
}

resource "aws_vpc" "matt-vpc" {
    cidr_block = "10.4.2.0/24"
    enable_dns_hostnames = true
    tags {
        Name = "matt-vpc"
    }
}
```

SImple little error about unable to resolve the host would be nice.

**Comment by radeksimko** [comment]
Hi,
can you confirm this is duplicate of https://github.com/hashicorp/terraform/issues/2202 ?

That fix in upstream probably isn't part of `0.5.3` build, but will be part of `0.6.0`, so you can either try building from source to see if it solves the problem or wait for the release.

**Comment by mitchellh** [comment]
This is certainly a duplicate. Thanks @radeksimko
